ansible/content-servers.yml

```diff
@@ -0,0 +1,197 @@
+    nginx_remove_default_vhost: true
+    nginx_vhosts:
+      - listen: "80"
+        server_name: "{{ inventory_hostname }}"
+        filename: "{{ inventory_hostname }}.80.conf"
+        extra_parameters: |
+          {{ nginx_shared_config }}
+
+      - listen: "443 ssl http2"
+        server_name: "{{ inventory_hostname }}"
+        filename: "{{ inventory_hostname }}.443.conf"
+        # Config partially taken from https://ssl-config.mozilla.org/#server=nginx&version=1.14.2&config=intermediate&openssl=1.1.1d&guideline=5.6
+        extra_parameters: |
+          ssl_certificate     /etc/letsencrypt/live/{{ inventory_hostname }}/fullchain.pem;
+          ssl_certificate_key /etc/letsencrypt/live/{{ inventory_hostname }}/privkey.pem;
+          ssl_session_timeout 1d;
+          ssl_session_cache shared:MozSSL:10m;  # about 40000 sessions
+          ssl_session_tickets off;
+          ssl_protocols TLSv1.2 TLSv1.3;
+          ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384;
+          ssl_prefer_server_ciphers off;
+
+          {{ nginx_shared_config }}
+
+  pre_tasks:
+    - name: Update & upgrade & cleanup apt.
+      apt:
+        update_cache: true
+        upgrade: safe
+        autoremove: yes
+        autoclean: yes
+        cache_valid_time: 86400  # a day is plenty
+      changed_when: false
+
+    - name: Install apt https support & cron.
+      apt:
+        name:
+          - apt-transport-https
+          - cron  # required for certbot
+        state: present
+
+  roles:
+    - geerlingguy.certbot
+    - geerlingguy.nginx
+
+  tasks:
+    - name: Flush handlers in case any configs have changed.
+      meta: flush_handlers
+
+    # Tests basic nginx
+    - name: Test secure connection to SSL domain.
+      uri:
+        url: https://{{ inventory_hostname }}/healthz
+        status_code: 200
+      delegate_to: localhost
+      become: false
+
+    # Tests base path
+    - name: Test secure connection to SSL domain.
+      uri:
+        url: https://{{ inventory_hostname }}/
+        status_code: 404
+      delegate_to: localhost
+      become: false
+
+    - name: Install some helpful utilities.
+      apt:
+        name:
+          - bash-completion
+          - logrotate
+          - molly-guard
+          - rsync
+          - sshguard
+          - unattended-upgrades
+          - vim
+        state: present
+
+    - name: Copy sshd config.
+      copy:
+        src: files/sshd_config
+        dest: /etc/ssh/sshd_config
+      notify:
+        restart ssh
+
+    - name: Create sshguard blacklist db directory.
+      file:
+        path: /var/db/sshguard
+        state: directory
+
+    # By default, sshguard listens to the entire journal... which includes itself. It gets a bit confused. Add -t sshd to stop it being silly
```

Comment:
This comment confuses me. I assume `-t sshd` makes it only listen to the ssh logs?

Reply:
It does. Previously it was listening to the whole log, which picked up itself. So any invalid login attempt was being immediately blocked. Which is probably undesirable for genuine cases. I've linked the actual debian bug report about it